core-bundle/src/Security/Authentication/Token/TokenChecker.php

```diff
@@ -96,11 +119,11 @@ public function isPreviewMode(): bool
 
     private function getToken(string $sessionKey): ?TokenInterface
     {
-        if (!$this->session->isStarted() || !$this->session->has($sessionKey)) {
-            return null;
-        }
+        $token = $this->getTokenFromStorage(substr($sessionKey, 10));
```

Comment:
Why did you add `substr($sessionKey, 10)` here? Should we not throw an `InvalidArgumentException` if the session key is too long?

Reply:
Its not about the length, the session key contains more that the necessary firewall context.

Reply:
I see. We should document this then I guess. 😆

Reply:
We should actually never have added the session kex constants but use the scopes…

Reply:
Is it too late to change it?

Reply:
well we could remove the use of it, but the constants _could_ be used by people? Not sure if they are API?

Reply:
We don't need to remove them. We can deprecate them and stop using them.

Reply:
So, does it make sense to hardcode this to `_security_` then?

Reply:
yes, it's hardcoded in Symfony as well: https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/Security/Http/Firewall/ContextListener.php#L64